Sources/PackageGraph/RepositoryPackageContainer.swift

```diff
@@ -395,39 +400,41 @@ public class RepositoryPackageContainerProvider: PackageContainerProvider {
         on queue: DispatchQueue,
         completion: @escaping (Result<PackageContainer, Swift.Error>) -> Void
     ) {
-        // If the container is local, just create and return a local package container.
-        if package.kind != .remote {
-            return queue.async {
-                let container = LocalPackageContainer(
+        if let repositorySpecifier = package.repositorySpecifier {
```

Comment:
@mattt here you would need to add a case for the registry, and probably we can rename RepositoryPackageContainerProvider to something more generic that deals with all PackageContainer types, perhaps rename to DefaultPackageContainerProvider